### PR TITLE
FEAT: Added PDF Converter for Prompt-to-PDF Conversion

### PR DESCRIPTION
**Pull request by Pratyaksh0215**
This PR introduces a new PDF Converter feature that converts text prompts into a PDF file. The converter supports both templated and non-templated PDF generation using Jinja2 for dynamic templating and FPDF for PDF creation. The feature addresses the need for converting textual prompts into formatted PDFs, which can be used in various PyRIT orchestrations.

Key highlights of the PR include:

Support for dynamic templates: If a base file with placeholders is provided, the prompt is embedded into the template using Jinja2.
Plain PDF generation: If no template is provided, the text prompt is directly embedded into a simple PDF.
Flexibility: The converter is designed to handle a variety of prompt inputs and future enhancements.
Related discussions: #423

Tests and Documentation
Unit tests have been added to validate the converter's functionality for both templated and non-templated PDF generation.
The following tests were performed:
Prompt injection into a dynamic PDF template.
Generating a PDF without a template.
Tests utilize mock prompts and template files to ensure correctness.
Jupytext: The integration tests were executed via Jupytext to verify that the new PDF converter integrates correctly with PyRIT orchestrations.

For more information on running tests and the setup, see [PyRIT Contribution Guide](https://github.com/Azure/PyRIT/tree/main/doc/contributing/tests.md)​

### CODE REVIEW

**Review by u7780339 — COMMENTED** [review]
Good work, just need some convention fix. The docs are usually used to show how to use your converter, so you might not want to create your PR to the main branch with the current docs

File: pyrit/prompt_converter/__init__.py [hunk 1]
```diff
@@ -88,4 +88,5 @@
     "UnicodeSubstitutionConverter",
     "UrlConverter",
     "VariationConverter",
+    "PDFConverter"
```
Comment (u7702792) [inline pyrit/prompt_converter/__init__.py:91]: I think they list the converter name in alphabet, maybe you can move it from the end to somewhere between "O" and "Q"

**Comment by u7702792** [comment]
Also, you might want to upload a test class for your converter if possible

File: pyrit/prompt_converter/pdf_converter.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            except FileNotFoundError:
+                logger.error(f"Template file {self.template_path} not found.")
+                raise
+        else:
+            content = prompt
+
+        # Create the PDF
+        pdf = FPDF()
+        pdf.add_page()
+        pdf.set_font("Arial", size=12)
```
Comment (u7702792) [inline pyrit/prompt_converter/pdf_converter.py:52]: maybe set the font as a parameters can allow users make adjustments based on their needs

**Pull request merged**